http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/DataprocInterpreter.scala

```diff
@@ -435,9 +442,65 @@ class DataprocInterpreter[F[_]: Timer: Parallel: ContextShift](
   def removeClusterIamRoles(googleProject: GoogleProject, serviceAccountInfo: WorkbenchEmail): F[Unit] =
     updateClusterIamRoles(googleProject, serviceAccountInfo, createCluster = false)
 
+  // Called at Leo boot time. Sets up the Dataproc image user group if it does not already exist.
+  // Group membership is modified at Dataproc cluster creation time.
   def setupDataprocImageGoogleGroup(implicit ev: Ask[F, AppContext]): F[Unit] =
```

Comment:
Just refactoring: previously this was split up across multiple methods, I feel it's easier to follow as a single for-comp.